### ISSUE TITLE
Invalid coords for TryGetPosition and TryGetRotation

### ISSUE THREAD

**Issue by hridpath** [Microsoft/MixedRealityToolkit-Unity#1290]
In the Unity.XR.WSA.Input.InteractionSourcePose the TryGet functions can return a true value with corrdinates in the results having a NaN value. I have created a Bug Report to Unity and waiting on the results.  This may be intentional on their part so inorder to prevent errors when using the output of these methods, A NaN check should be done prior to using the values.

As a Temporary fix the following code can be used allong with these TryGet functions.
Here is a snippit of code to fix this in teh MotionControllerVisualizer.cs 
![mcv_validpos_rot](https://user-images.githubusercontent.com/15804010/32352888-5ceb49bc-bfe0-11e7-892b-b2eef7b1916e.png)

**Comment by StephenHodgson** [comment]
I think there's a simpler way to check if the value is valid at all, including infinity

**Comment by hridpath** [comment]
TryParse by converting the value ToSting() would cover the infinity, but this can be very expensive in an Update. @StephenHodgson Do you think that Infinity would be invalid in this case with the Motion Controller?

**Comment by StephenHodgson** [comment]
Why wouldn't `float.IsInfinity(newPosition.y)` not work?

**Comment by keveleigh** [comment]
Why do we need to check IsInfinity?

**Comment by StephenHodgson** [comment]
Who know's.  We're getting NaNs so there is the possibility

**Comment by hridpath** [comment]
The IsInfinity check can be included in the return expression. Would you consider Infinity an invalid value to set a Position or Rotation??

**Comment by StephenHodgson** [comment]
@hridpath were you going to submit a PR for this?

**Comment by keveleigh** [comment]
Looks like GitHub didn't automatically close these issues when merged into master. I'll close these manually and look into why.
